### ISSUE TITLE
Determine Corresponding CSI Entries for Ping & Reply

### ISSUE THREAD

**Issue by TadiGuy** [dhalperi/linux-80211n-csitool-supplementary#224]
I have set up one machine as AP and another station/client. I can ping and collect CSI log on both . I'm trying to figure out corresponding CSI entries for each ping and reply at the station and AP. I understand one way to do this is to log payload information and look for 802.11 sequence number which requires writing a handler to process the payload information. Is that the only (and the best) way to go about this?

If so any hint on how to write such a handler will be helpful. I've traced read_bf_file.m and verified that my csi.dat contains both code 0xbb and 0xc1 for beamforming and payload respectively. I've spent some time studying read_bfee.c to use that as a reference but can't follow it well to understand how I would modify it to get payload info. I have also looked at figure 8-30 in the 802.11 standard which is of relavance. I'm still struggling a bit regarding how to write a handler to parse the data for payload info. A pseudocode or what objects to manipulate will be a good start. 

 If there is an alternative setup for accomplishing what I need with csi tool please let me know. Thanks.

**Comment by TadiGuy** [comment]
@dpward @dhalperi

**Comment by TadiGuy** [comment]
I hope you can chime in and give me some pointers. Thanks.

**Comment by swashah** [comment]
@TadiGuy : Were you able to log payload information? As I am also interested in it as I want to synchronize data of 02 devices. Thanks

**Comment by TadiGuy** [comment]
No, I haven't been able to synchronize data. I still need to do this to
finish my data collection. If you have any ideas may be we can collaborate
to figure it out.

**Comment by swashah** [comment]
@TadiGuy : I came across a modified version of tool which claims to be helpful but I could not set this up and gives error while generating binaries. 
you can have a look at 'https://github.com/jiangzp/linux-80211n-csitool-supplementary'
please let me know if you succeed in setting this up.

**Comment by swashah** [comment]
@TadiGuy : Did you get any chance to setup this modified version, as this could be helpful for sync.

**Comment by TadiGuy** [comment]
No, not yet. I first need to setup up monitor mode on another test bed
without disturbing my current AP-Station test bed. After that I'll proceed
to try this.

**Comment by swashah** [comment]
@TadiGuy : I have slight confusion in my experiment. May be you can have some idea. I have 02 laptops in monitor mode and third laptop is injecting packets. I control rate from transmitter and both the receiving laptops are using same antenna. I have kept the distance between 02 antennas less than half wavelength so logically there CSI (amplitude) should be similar (although misalignment could be there due to packet loss). But when I see CSI amplitude in matlab , data lacks similarity. Do you have any idea ?

**Comment by TadiGuy** [comment]
@swashah: Unless you've been able to solve the original problem both you
and I are seeking a solution (i.e. you an tell exactly what received
packets correspond to what transmitted packets) I think it will be hard to
know if the CSI you are comparing between the two antennas at the monitors
were received from the same injected packet. Are you able to align the
received packets with the transmitted packets?

Also I don't think you can count on the CSI amplitudes being similar
because there are many variables outside your control. Even with MIMO
receive antennas I don't see the same CSI amplitudes on all three antennas.

PM: tadiguy812-at-gmail-dot-com so that we don't clutter the thread with
unrelated comments. I'm still hoping that @dhalperi can provide some
guidance on my original post of identifying received packets.
